### DISCUSSION TITLE
Is there a way to make @updatedAt default value null?

### DISCUSSION THREAD

**Discussion by francismanansala** [prisma/prisma#11559] [Q&A]
I'm working on a project where we want our PostgreSQL DB to store updatedAt Timestamps with a default value of null. I am creating our model in our prisma schema with the property `updatedAt DateTime? @updatedAt` and when I use the prisma client to create a row for the table, it puts the current timestamp instead of null. Is there a way to have it created with updatedAt as null without explicitly sending `updatedAt` as null to the prisma client during creation? We would like this behaviour for all of our updateAt columns in the future.

Here are some example of I'm trying to do:

schema.prisma:
```
model User {
  id            Int       @id @default(autoincrement())
  email         String    @unique
  name          String
  createdAt     DateTime  @default(now())
  updatedAt     DateTime? @updatedAt
}
```

prisma client usage:
```
  prisma.user.create({
    data: {
      email: 'test@example.com'
      name: `first last`
    }
  })
```

result in DB:
![image](https://user-images.githubusercontent.com/10519099/152185665-6db97966-fddc-40a4-a3eb-c1ef7871df02.png)

We do not want to always explicitly set updatedAt to null when using prisma client, `prisma.user.create`. We'd like to be able to do something like changing the schema.prisma to have `@default(null)` for `updatedAt`

**Comment by seth** [comment]
How would you use the default null value and what do you gain vs having updatedAt set to NOW() on at time of creation?

**Reply by francismanansala** [reply]
We'd like to be able to tell easily if a row has never been updatedAt if it is null. It's for clarity: createdAt is when a recorded was added, updatedAt is when it was changed.

**Reply by jonahallibone** [reply]
Couldn't you compare the two for equality? If createdAt === updatedAt then it's never been updated

**Reply by francismanansala** [reply]
Thanks for the suggestion. We're not against the idea of doing `createdAt === updatedAt`, but we're aiming for data clarity and would like to make our columns unrelated as much as possible. If we were to do `createdAt === updatedAt`, they become dependant on each other and there's assumptions that prisma and our DB always use the same timestamp for multiple columns when a row is created for the prisma schema attributes `@default(now())` and `@updatedAt`.

**Comment by austincrim** [comment]
Hey @francismanansala 👋,

This sounds like a good idea! Would you mind opening a [feature request](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) with these details so our engineering teams can take a look at it and prioritize?

FYI, the `createdAt === updatedAt` workaround mentioned above is not 100% reliable and may cause bugs down the road.

**Reply by francismanansala** [reply]
Thanks for getting back to me @austincrim. I'm sorry my response was delayed.

I've created the feature request as you requested. Please let me know if I need to provide anymore information on the request: https://github.com/prisma/prisma/issues/12019

**Comment by vaintech** [comment]
I have same problem and thought it is because i did something wrong but discovered it's how Prisma works. what i did before was to set it null in base CreateDto class and inherit from it for each dto, i use prisma with nestjs and that's worked for me

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!

**Reply by Xevion** [reply]
Hello, I still think this discussion is relevant, and I would ask that it be re-opened. Thank you.
